lib/experimental/Information/Avatars/BaseAvatar/index.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  color as AvatarColors,
+  Avatar as AvatarComponent,
+  AvatarFallback,
+  AvatarImage,
+} from "@/ui/avatar"
+import { ComponentProps, forwardRef } from "react"
+
+type ShadAvatarProps = ComponentProps<typeof AvatarComponent>
+
+type Props = {
+  type: ShadAvatarProps["type"]
+  name: string | string[]
+  src?: string
+  size?: ShadAvatarProps["size"]
+  color?: ShadAvatarProps["color"] | "random"
+}
+
+function getInitials(
+  name: string | string[],
+  size?: ShadAvatarProps["size"]
+): string {
+  const nameArray = Array.isArray(name) ? name : [name]
+  const hasSmallSize = size === "xsmall" || size === "small"
+
+  if (hasSmallSize) return nameArray[0][0].toUpperCase()
```

Comment:
Surprised that this isn't making typescript complain, as `nameArray[0]` can be undefined. Or is it?

Reply:
somehow it thinks it will get a string. I added both the test and the fix for this case